DLSv2/Core/ManagedVehicle.cs

```diff
@@ -192,17 +189,17 @@ public void RegisterInputs()
                     {
                         ControlsManager.PlayInputSound();
                         int index = cG.Modes.IndexOf(mode);
-                        if (this.LightControlGroups[cG.Name].Item1 && this.LightControlGroups[cG.Name].Item2 == index)
-                            ControlGroupManager.ToggleControlGroup(this, cG.Name);
+                        if (LightControlGroups[cG.Name].Enabled && LightControlGroups[cG.Name].Index == index)
+                            LightControlGroups[cG.Name].Toggle();
                         else
-                            ControlGroupManager.SetControlGroupIndex(this, cG.Name, index);
-                        LightController.Update(this);
+                            LightControlGroups[cG.Name].Index = index;
+                        UpdateLights();
                     };
                 }
             }
 
             // Audio Control group and modes keys
-            foreach (AudioControlGroup cG in AudioControlGroupManager.ControlGroups[Vehicle.Model].Values)
+            foreach (AudioControlGroup cG in AudioControlGroups.Values.Select(x => x.BaseControlGroup))
```

Comment:
Same thing here: loop through the wrappers, not the base control groups, then use the wrappers directly